### ISSUE TITLE
More Force Merge Threads by default 

### ISSUE THREAD

**Issue by gbbafna** [opensearch-project/OpenSearch#7832]
**Is your feature request related to a problem? Please describe.**

Today Force Merge threads are always 1 per node . Even if the nodes have lots of CPU, the number of force merge threads is limiting . 

**Describe the solution you'd like**

Use `ScalingExecutorBuilder` instead of `FixedExecutorBuilder` which caps the max threads based on number of processors 

**Describe alternatives you've considered**

Use yaml to update the count of Force Merge threads .

**Comment by shwetathareja** [comment]
@gbbafna The force merges can interfere with the active write/ read traffic if no. of threads are increased. Do you have plan to benchmark?

**Comment by gbbafna** [comment]
Yes Shweta, we will definitely need to do benchmarks on this one.
